### PR TITLE
Ocpp20 config impl

### LINKED ISSUE #100: 092-[Back] Complete configuration action in 2.0.1
After process in 1.6 is done, adapt this to 2.0.1 version. 

Marius DONNE said it won't take more than 2h

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1206575137133723/1206637549836846) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by tvelu77**
The server is now capable of sending configurations and receiving response in OCPP 2.0

### CODE REVIEW

File: src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java [hunk 1]
```diff
@@ -21,13 +22,16 @@ default MessageTypeRequest messageName() {
 
   long messageId();
 
+  boolean isRequest();
+
   /**
    * Defines the message type which the server could receive from the chargepoint.
```
Comment (Idhao) [inline src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java]: better javadoc for this enum

File: src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java [hunk 2]
```diff
@@ -94,6 +99,7 @@ static Optional<WebSocketMessage> parse(String message) {
     return switch (MessageType.codeToEnum(callType)) {
       case RESPONSE -> {
         try {
+          array = message.substring(1, message.length() - 1).split(",", 3);
```
Comment (Idhao) [inline src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java:133]: should deserve an technical explanation ex : why split 3 ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/ocpp/ocpp16/OcppConfigurationObserver16.java [hunk 3]
```diff
@@ -100,7 +100,12 @@ private void processBootNotification(
     var config = currentChargepoint.getConfiguration();
     if (status.getStatus() == Status.StatusProcess.FINISHED
             && config.getLastEdit().before(status.getLastUpdate())) {
-      // TODO : Log, the chargepoint is ready !
```
Comment (Idhao) [inline src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/ocpp/ocpp16/OcppConfigurationObserver16.java:103]: todo should stay ?

**Review by Idhao — COMMENTED** [review]
waiting your modifs

File: src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java (hunk 1 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java]: Done

[808e6e5](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/140/commits/808e6e51e4d32f2bb3e83b79ab830b933d6565ee)

File: src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java (hunk 2 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java:133]: Done

[a2c471b](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/140/commits/a2c471b5e927284e9d02f4a1c97a2388d714406d)

File: src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/ocpp/ocpp16/OcppConfigurationObserver16.java (hunk 3 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/ocpp/ocpp16/OcppConfigurationObserver16.java:103]: Done

[ed79c97](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/140/commits/ed79c97bda9a45dee4d0c669dad6450f0dd575cc)

File: src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java [hunk 4]
```diff
@@ -14,18 +14,47 @@
...
   String data();
 
+  /**
+   * Returns the message's type.<br>
+   * By default, it returns null when the message is not a request.
+   *
+   * @return {@link MessageTypeRequest}.
+   */
   default MessageTypeRequest messageName() {
     return null;
```
Comment (Idhao) [inline src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java:34]: null and nothing else ? are you okay ?

**Review by Idhao — CHANGES_REQUESTED** [review]
if the null part is volontary, then you can merge

File: src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java (hunk 4 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepointwebsocket/WebSocketMessage.java:34]: Totally normal, by default, we consider the message as a Response or something other than a request (it doesn't have a message type).
In the code, it is perfectly handled as it

**Comment by tvelu77** [comment]
@Idhao Requested an another review as the null part is volontary

**Review by Idhao — APPROVED** [review]
I understand now, ty incr doc

**Pull request merged**